### PR TITLE
Use `orientation` model source property to update the 3D puck's bearing.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

Use `orientation` model source property to update the 3D puck's bearing, as it is more efficient than updating the layer property.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt [hunk 1]
```diff
@@ -42,8 +42,12 @@ internal class ModelSourceWrapper(
     }
   }
 
-  fun setPosition(value: List<Double>) {
-    val property = hashMapOf(Pair(POSITION, Value(value.map { Value(it) })), Pair(URL, Value(url)))
+  fun setPositionAndOrientation(latLng: List<Double>, orientation: List<Double>) {
+    val property = hashMapOf(
+      Pair(POSITION, Value(latLng.map { Value(it) })),
```
Comment (yunikkk) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:47]: nit : `map(::Value)` should be possible

Comment (alexander-kulikovskii) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:47]: how about small extension for `List<Any_type_for_value>`
```
fun List<T>.mapToValue():Value = Value(this.map { item -> Value(item) })
```

Comment (alexander-kulikovskii) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:47]: nit:
```suggestion
      POSITION to Value(latLng.map { Value(it) }),
```

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelLayerRenderer.kt [hunk 2]
```diff
@@ -73,14 +80,25 @@ internal class ModelLayerRenderer(
...
+
+  private fun updateLocationOrBearing(point: Point? = null, bearing: Double? = null) {
+    point?.let {
+      lastLocation = it
+    }
+    bearing?.let {
+      lastBearing = it
+    }
+    lastLocation?.let { location ->
+      val latLng = listOf(location.longitude(), location.latitude())
```
Comment (alexander-kulikovskii) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelLayerRenderer.kt:98]:
```suggestion
      val latLngAsList = listOf(location.longitude(), location.latitude())
```

Comment (alexander-kulikovskii) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelLayerRenderer.kt:98]: should it be `latLng` or `lngLat` ?

**Comment by yunikkk** [comment]
@pengdev could you elaborate more on why source property is faster than model one?

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt (hunk 1 above)
Comment (pengdev) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:47]: good tips

**Comment by pengdev** [comment]
> @pengdev could you elaborate more on why source property is faster than model one?

@yunikkk this is the feedback from gl-native folks, generally updating the data source is faster than updating a layer property. This also aligns with the implementation in the iOS SDK.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelLayerRenderer.kt (hunk 2 above)
Comment (pengdev) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelLayerRenderer.kt:98]: yes, it should be lngLat, fixed.

File: plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt [hunk 3]
```diff
@@ -42,9 +42,13 @@ internal class ModelSourceWrapper(
...
   }
 
-  fun setPosition(value: List<Double>) {
-    val property = hashMapOf(Pair(POSITION, Value(value.map { Value(it) })), Pair(URL, Value(url)))
-    val updateModel = hashMapOf(Pair(DEFAULT_MODEL_NAME, Value(property)))
+  fun setPositionAndOrientation(lngLat: List<Double>, orientation: List<Double>) {
+    val property = hashMapOf(
+      POSITION to Value(lngLat.map(::Value)),
+      ORIENTATION to Value(orientation.map(::Value)),
+      URL to Value(url)
```
Comment (Chaoba) [inline plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:49]: It's a litter wired to update URL in this method, is it needed here?

Comment (pengdev) [reply plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/ModelSourceWrapper.kt:49]: We need to update the source property in a bundle, and a url is required here.

**Review by ank27 — APPROVED** [review]

**Review by Chaoba — APPROVED** [review]

**Pull request merged**